### DISCUSSION TITLE
RFE: save as LaTeX input/tikz would be good option to have?

### DISCUSSION THREAD

**Discussion by maitra** [xournalpp/xournalpp#3038] [Ideas]
Hi, it would be wonderful if we could have an option to save as LeTeX input file, perhaps with tikz or something else. Is this possible to do?

**Comment by mjg** [comment]
> Hi, it would be wonderful if we could have an option to save as LeTeX input file, perhaps with tikz or something else. Is this possible to do?

Do you mean to save (export) in a format which LaTeX (using e.g. the graphics page) can input/include?

**Comment by maitra** [comment]
> > Hi, it would be wonderful if we could have an option to save as LeTeX input file, perhaps with tikz or something else. Is this possible to do?
> 
> Do you mean to save (export) in a format which LaTeX (using e.g. the graphics page) can input/include?

I think that this option is to some extent already possible. I am thinking of having a LaTeX extension which creates LaTeX document that can then be modified. This would allow us the possibility to include handwritten documents as LaTeX code. xfig has the ability, but does not allow for hand-drawn options.

**Comment by mjg** [comment]
I still don't understand how what is possible is different from what you want. xfig can generate various output formats that you can then include in LaTeX, yes, and so can xournalpp. PDF is a nice scalable format for that, you may just need to crop it before including it in LaTeX. Also, this is much more efficient than anything in tikz...

You could convert the svg export of xournalpp to tikz, but what for? You'd be much better off editing the xopp file than the generated svg or tikz if you want to do some manual editing or post processing. Also note that you can render LaTeX code within xournalpp for a quick label or such.

**Comment by maitra** [comment]
So, the main benefit is more flexibility. I am thinking that I have a complicated LaTeX document but I want to add some handwritten things in it, along with geometric shapes, but I want to change fonts, etc along the way. It can be done as you say calling in LaTeX code, but it is more complicated. Besides journals would not accept a xopp document for publication. Btw I did not know that we can convert the  svg export of xournalpp to tikz. I will look into that.

**Comment by mjg** [comment]
Hmm, what is complicated about `\includegraphics{handwritten}` when you have exported from xournalpp to `handwritten.pdf`? This does not keep you from changing anything you want in your LaTeX file, including fonts. I guess I still don't know what exactly you are trying to do. Probably the only thing that `svg2tikz` buys you is: you can produce a monolithic LaTeX file which contains the tikz code that reproduces the xournalpp, very slowly, unless you use luatex. Journals should be happy about "publication ready figures" in PDF format, by the way.

**Comment by maitra** [comment]
I want to be able to modify, using LaTeX, the graphics inside the "handwritten", so for example if I want to change font, etc, I do not have to go back and edit everything again.  I m hoping to get the full power of TikZ for annotating, manipulating, and integrating with LaTeX. 

I will look into the `svg2tikz`, thanks again! [](https://github.com/xyz2tex/svg2tikz)
